types/user.hh

```diff
@@ -47,6 +47,21 @@ public:
     sstring get_name_as_string() const;
     sstring get_name_as_cql_string() const;
 
+    /* Returns set of user-defined types' names referenced by this UDT
+     * 
+     * Example:
+     * create type some_udt {
+     *   a frozen<udt_a>,
+     *   m frozen<map<udt_a, udt_b>,
+     *   t frozen<tuple<int, list<udt_c>>   
+     * }
+     * get_all_referenced_user_types() will return {udt_a, udt_b, udt_c}.
+     *
+     * NOTE: UDTs used to create a new one must be in the same kespace,
+     * so the functions returns only names.
```

Comment:
Is it true? Can't I refer to a name from another keyspace?

```g
userTypeName returns [uninitialized<cql3::ut_name> name]
    : (ks=ident '.')? ut=non_type_ident { $name = cql3::ut_name(ks, ut); }
    ;
```

Reply:
Although CQL grammar allows it, it is not possible. We have a runtime check:

```c++
cqlsh> create type ks1.x (a int);
cqlsh> create type ks2.x (a frozen<ks1.x>);
InvalidRequest: Error from server: code=2200 [Invalid query] message="Statement on keyspace ks2 cannot refer to a user type in keyspace ks1; user types can only be used in the keyspace they are defined in"
cqlsh> 

```